test/fail_compilation/deprecate16597.d

```diff
@@ -0,0 +1,15 @@
+// REQUIRED_ARGS: -de
+/* TEST_OUTPUT:
+---
+fail_compilation/deprecate16597.d(9): Deprecation: variable `deprecate16597.aa1` associative arrays are not thread safe and cant be declared `shared`.Use `__gshared` and barriers instead
+fail_compilation/deprecate16597.d(10): Deprecation: variable `deprecate16597.aa2` associative arrays are not thread safe and cant be declared `shared`.Use `__gshared` and barriers instead
+---
+*/
+// https://issues.dlang.org/show_bug.cgi?id=16597
+shared int[int] aa1;
+shared(int[int]) aa2;
```

Comment:
We should probably also disallow `shared(V)[K]`, as I don't think the AA implementation takes any precautions to guard reads/writes of shared values. Can you add a test for this as well?